### DISCUSSION TITLE
why use constexpr function obj instead of normal function?

### DISCUSSION THREAD

**Discussion by netcan** [actor-framework/actor-framework#1210] [Q&A]
https://github.com/actor-framework/actor-framework/blob/e754ee261aee36cfb3925f1b6985c75946f2e363/libcaf_core/caf/detail/behavior_impl.hpp#L195

what dose constexpr funtion obj mean? why not use normal funtion as below?
```cpp
  template <class... Ts>
  constexpr auto make_behavior(Ts... xs)  {
    if constexpr ((is_timeout_definition<Ts>::value || ...)) {
      auto args = std::tie(xs...);
      auto& tdef = std::get<sizeof...(Ts) - 1>(args);
      behavior_factory_t<std::decay_t<decltype(tdef)>> f{tdef};
      std::make_index_sequence<sizeof...(Ts) - 1> indexes;
      return detail::apply_args(f, indexes, args);
    } else {
      using type = default_behavior_impl<std::tuple<Ts...>>;
      dummy_timeout_definition dummy;
      return make_counted<type>(std::make_tuple(std::move(xs)...), dummy);
    }
  }
```

**Comment by Neverlord** [comment]
Because you cannot pass around a "raw" template function in C++. Here's an example that wouldn't work with a function: https://github.com/actor-framework/actor-framework/blob/e754ee261aee36cfb3925f1b6985c75946f2e363/libcaf_core/caf/blocking_actor.hpp#L378.
